src/isDirectory.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+'use strict';
+
+function isDirectory(fs: FS, filepath: string): Promise<boolean> {
+  if (typeof filepath !== 'string') {
+    return Promise.reject(new TypeError('expected filepath to be a string'));
+  }
```

Comment:
Curious: Why do we need this assertion? We aren't making similar assertions elsewhere, right?

Reply:
I referred the `is-directory` implementation and only modified it to return a `Promise`.

Reply:
Ah, well I think we should remove this since we have Flow in place and aren't asserting against argument types elsewhere.

Reply:
Alright, done.